src/main/kotlin/AppModule.kt

```diff
@@ -47,49 +50,65 @@ fun Application.module() {
         }
         route("/") {
             get {
-                call.respond(ThymeleafContent("index", mapOf("obj" to "")))
+                call.respond(ThymeleafContent("index", mapOf("obj" to ""), locale = Locale("vi")))
             }
-            post {
-                val data = call.receive<FilePathTestModel>()
-                val serializedResult = processFileUpload(data)
+        }
+        post("/upload") {
+            val multiPart = call.receiveMultipart()
+            val dataMap: MutableMap<String, String> = mutableMapOf()
+            val fileUploadModel: FileUploadModel
+            var file = File("")
```

Comment:
maybe it's better to not declare this here? and just declare file as a val variable on line 68 instead.

Reply:
I tried that but that "file" variable should be out of the scope of multiPart.forEachPart block.

Reply:
maybe that will change if you move things out into a different class. You could store the file as a class property and access it that way? It would probably make things cleaner overall